### DISCUSSION TITLE
Ever thought  to build the docker container for ARM / Raspberry Pi?

### DISCUSSION THREAD

**Discussion by hahusu** [tobybatch/kimai2#464] [Ideas]
Hi there, 

I would love to selfhost KIMAI as Docker Container on my local pi for time measuring my construction work tasks at home.
Is there any Docker Image that work with ARM / Raspbian?

I currently dont know what steps it take to bring it on the Pi but i guess if there is nothing planned nor existing i will try my best to bringt it on a ARM based docker image myself.

Finally i just want to let you know you did a great Job with Kimai (and also with the containers) which work like a charm (on non ARM based machines)

Thank you very much!  PS: (Where can i buy you guys a coffee?)

**Comment by tobybatch** [comment]
You could look at this: https://github.com/tobybatch/kimai2/discussions/465

I did have a go but I'm not sure if the buildx in github action pushes a multi arch version. Have you tried pulling it for arm? 

If we don't build for arm then I'll look into it again, we really should support ARM CPUs

And look [here](https://github.com/sponsors/tobybatch?frequency=one-time&sponsor=tobybatch) for sponsorship, that would be great. Thanks

**Comment by hahusu** [comment]
Hi Toby, 

thank you very much! The build you  mentioned (multi-arch branch 1.8 prod) in #465 worked like a charm.
Now the only thing that keeps me away from using it, is sthat i get a 500 Internal Server Error.

![image](https://user-images.githubusercontent.com/11199338/217344614-b25a97d7-3ad2-42cc-af6c-b4c0d245fa81.png)

![image](https://user-images.githubusercontent.com/11199338/217345086-6a5b625f-11ea-4d2c-8ea4-fa368cb7ef21.png)

I think the reason is the fact that i use the local network Ip of the Server to call the Kimai Webserver.

Edit:
I think that should do the trick 👍 
![image](https://user-images.githubusercontent.com/11199338/217446064-42ed35e5-a843-486d-8030-e2b0ec5f5a3c.png)


Best regards and also thank you for the sponsorship link i will send a couple of beer's to you guys!

Thank you!

**Comment by tobybatch** [comment]
Yep that's probably it. 

On another note I had a quick look at enabling multi arch builds but the blocker is that symfony doesn't seem to support ARM7.

You can build your own images:

 * Install buildx https://github.com/docker/buildx
 * Build the image:
     * https://tobybatch.github.io/kimai2/build.html
     * BUT use the buildx command: `docker buildx build --platform linux/arm64/v8,linux/amd64 -t kimai/kimai2:multi .`

I'll try and enable this in the github actions, but for now the above should work.

**Reply by kevinpapst** [reply]
> symfony doesn't seem to support ARM7.

Really? Where did you read that? Symfony is a PHP framework and should run wherever PHP runs. At least I hope so :)

**Reply by hahusu** [reply]
Thank you very much for your buildx hints. I'll give it a try next days - for future builds.
For now i am happy with the provided solution (multi-arch branch 1.8 prod)

Everything i wished works fine now on mi raspberry Pi.👍

**Comment by tobybatch** [comment]
This command is failing on the build:

wget https://github.com/symfony-cli/symfony-cli/releases/latest/download/symfony-cli_linux_arm.tar.gz

**Reply by kevinpapst** [reply]
Ah, you mean the CLI. Maybe just a wrong filename?

<img width="1688" alt="Bildschirm­foto 2023-02-09 um 12 46 15" src="https://user-images.githubusercontent.com/533162/217804282-5ded9ea8-75f2-4d70-b019-847b0155b058.png">

https://github.com/symfony-cli/symfony-cli/releases/download/v5.4.21/symfony-cli_linux_arm64.tar.gz

**Reply by tobybatch** [reply]
I didn't specify the filename. I'll investigate later, the ARM8 version builds fine.

Fails:

    docker buildx build --platform linux/arm/v7,linux/arm64/v8,linux/amd64 -t kimai:kimai2:multi . 

Succeeds:

    docker buildx build --platform linux/arm64/v8,linux/amd64 -t kimai/kimai2:multi .

**Reply by kevinpapst** [reply]
Why do we need the symfony-cli? Maybe we can replace it completely?

**Reply by tobybatch** [reply]
I'll look into that, it does not seem to be used anymore, well not referenced directly in the docker build process. It may left over from an older build process and never clean up. Bad Toby.
